### DISCUSSION TITLE
correctly deal with job retries on openpai

### DISCUSSION THREAD

**Discussion by QuanluZhang** [microsoft/nni#3769] [New Feature Design Discussion]
**Short summary about the issue/question**: a job on openpai may be retried. In the current version, nni is not aware of such event. This may induce potential issues, for example, an assessor may find such a trial's learning curve is strange, leading to incorrect behavior.

**Brief what process you are following**: normal

**How to reproduce it**: when a trial is retried on openpai

**nni Environment**:
- nni version: 0.5.2
- nni mode(local|pai|remote): pai
- OS: ubuntu
- python version: 3.5
- is conda or virtualenv used?: no
- is running in docker?: no

**Anything else we need to know**: 
Related to #863 and #865.

**Comment by leelaylay** [comment]
I think it is an important bug need to be fixed. Related to #863 and #865.
